VSRAD.DebugServer/ObservableProcess.cs

```diff
@@ -0,0 +1,162 @@
+﻿using System;
+using System.Diagnostics;
+using System.Text;
+using System.Threading.Tasks;
+using VSRAD.DebugServer.IPC.Responses;
+
+namespace VSRAD.DebugServer
+{
+    public sealed class ObservableProcess
+    {
+        public event EventHandler ExecutionStarted;
+        public event EventHandler<string> StdoutRead;
+        public event EventHandler<string> StderrRead;
+
+        private readonly Process _process;
+        private readonly bool _waitForCompletion;
+        private readonly int _timeout;
+        private bool _stoppedByTimeout = false;
+
+        public ObservableProcess(IPC.Commands.Execute command)
+        {
+            _process = new Process
+            {
+                StartInfo = new ProcessStartInfo(command.Executable, command.Arguments)
+                {
+                    WorkingDirectory = command.WorkingDirectory
+                },
+                EnableRaisingEvents = true
+            };
+            if (command.RunAsAdministrator)
+            {
+                _process.StartInfo.UseShellExecute = true;
```

Comment:
Are you sure you don't need to redirect output here? UseShellExecute with the value of true redirects it to the console.

Reply:
You're right, we don't currently capture the output when running commands as administrator because we launch them via cmd + `runas`, which takes care of the UAC prompt for us.

The problem is, the standard `Process` API (`RedirectStandardOutput`) cannot be used together with `UseShellExecute`
("The Process object must have the UseShellExecute property set to false in order to redirect IO streams.").

I'm sure we can find an alternative approach for running a process with eleavated privileges while redirecting its IO streams, but I'm afraid it will require using native APIs and, consequently, much more code. This is something that we probably should file a separate issue for.